odbc/src/api/get_diag_field_tests.rs

```diff
@@ -0,0 +1,348 @@
+use crate::{
+    api::{definitions::DiagType, errors::ODBCError},
+    handles::definitions::*,
+    SQLGetDiagFieldW,
+};
+use odbc_sys::{HandleType, SqlReturn};
+
+const UNIMPLEMENTED_FUNC: &str = "HYC00\0";
+
+mod unit {
+    use super::*;
+    use std::ffi::c_void;
+
+    fn validate_integer_diag_field(
+        handle_type: HandleType,
+        handle: *mut MongoHandle,
+        diag_identifier: DiagType,
+        expected_value: u64,
+    ) {
+        let diag_info_ptr = &mut 0u64 as *mut _ as *mut c_void;
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetDiagFieldW(
+                    handle_type,
+                    handle as *mut _,
+                    1,
+                    diag_identifier,
+                    diag_info_ptr,
+                    0,
+                    &mut 0
+                )
+            );
+            assert_eq!(expected_value, *(diag_info_ptr as *const u64));
+        }
+    }
+
+    fn validate_message_text(handle_type: HandleType, handle: *mut MongoHandle) {
+        const ERROR_MESSAGE: &str = "[MongoDB][API] The feature SQLDrivers is not implemented\0";
+        let message_text = &mut [0u16; 57] as *mut _ as *mut c_void;
+        let string_length_ptr = &mut 0;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetDiagFieldW(
+                    handle_type,
+                    handle as *mut _,
+                    1,
+                    DiagType::MessageText,
+                    message_text,
+                    57,
+                    string_length_ptr
+                )
+            );
+            assert_eq!(
+                ERROR_MESSAGE,
+                String::from_utf16(&*(message_text as *const [u16; 57])).unwrap()
+            );
+            assert_eq!(56, *string_length_ptr);
+        }
+    }
+
+    fn validate_sql_state(handle_type: HandleType, handle: *mut MongoHandle) {
+        let message_text = &mut [0u16; 6] as *mut _ as *mut c_void;
+        let string_length_ptr = &mut 0;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetDiagFieldW(
+                    handle_type,
+                    handle as *mut _,
+                    1,
+                    DiagType::SqlState,
+                    message_text,
+                    6,
+                    string_length_ptr
+                )
+            );
+            assert_eq!(
+                UNIMPLEMENTED_FUNC,
+                String::from_utf16(&*(message_text as *const [u16; 6])).unwrap()
+            );
+            assert_eq!(5, *string_length_ptr);
+        }
+    }
+
+    fn validate_return_code(handle_type: HandleType, handle: *mut MongoHandle) {
```

Comment:
I understand the idea that since you mapped it to SQL_SUCCESS, you want to make sure there is no side effect since the input/output values are not altered in any way. The comment before the assert is helpful in pointing this out, but require a little more thinking that it should be necessary I feel.
Could you add a comment at on top of the function explaining that this call is a sort of a no-op because we rely on the DM to handle it.
That would make it 100% clear.